tasks/main.yml

```
@@ -105,30 +80,32 @@
     state: present
     login_user: "{{ database_root_user }}"
     login_password: "{{ database_root_password }}"
+    check_implicit_admin: true
+    login_unix_socket: "{{ mariadb_socket_path }}"
 
-## configuration steps to mimic what 'mysql_secure_installation' does:
+## some configuration steps to mimic what 'mysql_secure_installation' does:
 
 - name: Delete anonymous user
   community.mysql.mysql_user:
     user: ""
-    host: "{{ item }}"
+    host_all: true
     state: absent
     login_user: "{{ database_root_user }}"
     login_password: "{{ database_root_password }}"
-  loop:
-    - "localhost"
-    - "{{ ansible_fqdn }}"
-  no_log: true
+    check_implicit_admin: true
+    login_unix_socket: "{{ mariadb_socket_path }}"
 
 - name: Ensure root user can only connect from the local machine
-  community.mysql.mysql_user:
-    user: "{{ database_root_user }}"
-    password: "{{ database_root_password }}"
-    host: "{{ item }}"
+  community.mysql.mysql_query:
+    query:
+      - DELETE FROM mysql.user WHERE User='root' AND Host NOT IN ('localhost', '127.0.0.1', '::1')
     login_user: "{{ database_root_user }}"
     login_password: "{{ database_root_password }}"
-  loop:
-    - "::1"
-    - "localhost"
-    - "{{ ansible_fqdn }}"
-  no_log: true
+    login_unix_socket: "{{ mariadb_socket_path }}"
+
+- name: Ensure unix socket as login method for root user
+  community.mysql.mysql_user:
+    name: "{{ database_root_user }}"
+    plugin: unix_socket
+    check_implicit_admin: true
+    login_unix_socket: "{{ mariadb_socket_path }}"
```

Enabling unix socket plugin for root user authentication will disallow password based authentication (see [docs](https://mariadb.com/docs/server/reference/plugins/authentication-plugins/authentication-plugin-unix-socket)). IMO there is no need to run this query. Isn't it?



```suggestion
- name: Ensure unix socket as login method for root user
  community.mysql.mysql_user:
    name: "{{ database_root_user }}"
    plugin: unix_socket
    check_implicit_admin: true
    login_unix_socket: "{{ mariadb_socket_path }}"
```